File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import PageResult.PageResult;
import com.pinyougou.pojo.TbBrand;
import PageResult.InsertResult;
import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有品牌
     * @return
     */
    public List<TbBrand> findAllBrands();

    /**
     * 分页查询所有品牌
     * @param PageNum 当前页
     * @param PageSize 每页显示条数
     * @return
     */
    public PageResult findByPages(int PageNum,int PageSize);

    /**
     * 添加一个品牌
     * @param tbBrand
     * @return
     */
    public InsertResult insertBrand(TbBrand tbBrand);

    /**
     * 根据id查询当前品牌，进行回显
     * @param id
     * @return
     */
    public TbBrand findOne(Long id);

    /**
     * 修改当前品牌
     * @param tbBrand
     * @return
     */
    public InsertResult updateBrand(TbBrand tbBrand);

    /**
     * 根据ids数组，删除品牌们
     * @param ids
     * @return
     */
    public void deleteBrands(Long[] ids);

    /**
     * 模糊查询，并分页
     * @param tbBrand
     * @param PageNum
     * @param PageSize
     * @return
     */
    public PageResult findIndistinct(TbBrand tbBrand,int PageNum,int PageSize);

    /**
     * 品牌下拉列框数据
     * @return
     */
    public List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//创建服务中心
app.service("brandService",function ($http) {
    //定义方法新增品牌,修改品牌
    this.insertBrand = function (entity) {
        return $http.post('../brand/insertBrand.do', entity);
    }

    //定义方法查询品牌
    this.findOne = function (id) {
        return $http.get('../brand/findOne.do?id='+ id);
    }

    //删除选中的ids
    this.deleIds = function (ids) {
        return $http.post("/brand/deleResult.do", ids)
    }

    //模糊查询
    this.findIndistinct = function (page,size,IndistinctEntity) {
        return $http.post('../brand/findIndistinct.do?page=' + page + '&size=' + size, IndistinctEntity);
    }

    //查询品牌下拉列表
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do?');
    }

});